runtime/src/snapshot_utils.rs

```diff
@@ -798,6 +1213,75 @@ fn sort_snapshot_archives(snapshot_archives: &mut Vec<SnapshotArchiveInfo>) {
     snapshot_archives.sort_unstable_by(|a, b| b.slot.cmp(&a.slot));
 }
 
+/// Get a list of the incremental snapshot archives in a directory
+fn get_incremental_snapshot_archives<P>(
+    snapshot_output_dir: P,
+) -> Vec<IncrementalSnapshotArchiveInfo>
+where
+    P: AsRef<Path>,
+{
+    match fs::read_dir(&snapshot_output_dir) {
+        Err(err) => {
+            info!(
+                "Unable to read snapshot directory: err: {}, path: {}",
+                err,
+                snapshot_output_dir.as_ref().display()
+            );
+            vec![]
+        }
+        Ok(files) => files
+            .filter_map(|entry| {
+                if let Ok(entry) = entry {
+                    let path = entry.path();
+                    if path.is_file() {
+                        if let Some((base_slot, slot, hash, archive_format)) =
+                            parse_incremental_snapshot_archive_filename(
+                                path.file_name().unwrap().to_str().unwrap(),
+                            )
+                        {
+                            return Some(IncrementalSnapshotArchiveInfo {
+                                inner: SnapshotArchiveInfo {
+                                    path,
+                                    slot,
+                                    hash,
+                                    archive_format,
+                                },
+                                base_slot,
+                            });
+                        }
+                    }
+                }
+                None
+            })
+            .collect(),
+    }
+}
+
+/// Get a sorted list of the incremental snapshot archives in a directory
+#[cfg(test)]
+fn get_sorted_incremental_snapshot_archives<P>(
+    snapshot_output_dir: P,
+) -> Vec<IncrementalSnapshotArchiveInfo>
+where
+    P: AsRef<Path>,
+{
+    let mut incremental_snapshot_archives = get_incremental_snapshot_archives(snapshot_output_dir);
+    sort_incremental_snapshot_archives(&mut incremental_snapshot_archives);
+    incremental_snapshot_archives
+}
+
+/// Sort the list of incremental snapshot archives, first by base (full snapshot) slot in
+/// descending order, then by "regular" (incremental snapshot) slot in descending order
+fn sort_incremental_snapshot_archives(
+    incremental_snapshot_archives: &mut Vec<IncrementalSnapshotArchiveInfo>,
+) {
+    incremental_snapshot_archives.sort_unstable_by(|a, b| {
```

Comment:
how about defining `Ord` for `IncrementalSnapshotArchiveInfo`, so that we can just call `sort()`?

Reply:
I actually purposely didn't want to do that, since I didn't feel there was a _universal_ ordering of this type that applied in all situations. We know in the context of the _callers_ of this function that they want the sorting to be first by base slot in descending order, then by "regular" slot in descending order, but maybe that's not the case for the next caller to come along.

What do you think? I can definitely change this to use `Ord` instead, as we know all the callers of this function.

Reply:
ooh, `a` and `b` is swapped in the code... 

could you at least comment it. I think it's too easy to miss this...

or better, I rather strive to keep to avoid descending order except performance sensitive code.

in this case, I'd write like this:

```rust
impl Ord for IncrementalSnapshotArchiveInfo {
   ... // COMMENT: yea, (base slot asc, regular slot asc) should be kind of universal order, assuming it's aligning with chronological order in the  snapshot lifecycle.
}

...


sort_incremental_snapshot_archives(&mut incremental_snapshot_archives); // COMMENT: or just call straightly: incremental_snapshot_archives.sort_unstable();


// pick the highest (i.e. latest one) one
incremental_snapshot_archives.into_iter().rev().next()
```

Reply:
OK, I've put `Ord` back in and reworked the code to use `sort_unstable()` directly.